sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1425,6 +1446,48 @@ pi_result _pi_context::getAvailableCommandList(
   return pi_result;
 }
 
+_pi_queue::pi_queue_group_t &_pi_queue::getQueueGroup(bool UseCopyEngine) {
+  if (Device->useImmediateCommandLists()) {
+
+    bool ThreadSpecificImmCmdLists = Device->useImmediateCommandLists() == 2;
+
+    if (ThreadSpecificImmCmdLists) {
+      // Thread id will be used to create separate imm cmdlists per thread.
+      auto TID = std::this_thread::get_id();
+      if (UseCopyEngine) {
+
+        std::pair<std::map<std::thread::id, pi_queue_group_t>::iterator, bool>
+            Result = CopyQueueGroupsByTID.insert({TID, CopyQueueGroup});
```

Comment:
What does it mean to use the single "CopyQueueGroup" for each TID?

Reply:
See new comments. The non-threaded Compute/Copy queue groups serve as initializers, (somewhat like OpenMP firstprivate initialization).

Reply:
Sorry, but I still don't understand that. Once you initialize it will stay, but we need different queue groups for different threads.